share/namespace.go

```diff
@@ -104,6 +131,17 @@ func ValidateUserNamespace(version uint8, id []byte) error {
 	return validateID(version, id)
 }
 
+// ValidateForData checks if the Namespace is of real/useful data.
+func ValidateForData(n Namespace) error {
```

Comment:
I wonder why we want to keep validators as functions rather than methods. Functions are harder to find when you work with namespaces, and usually, in the node, we want to validate them on the input. 

I guess it is more of a question to @cmwaters and @rootulp on why `ValidateUserNamespace` was made a func, while it could be `Namespace.ValidateUser` or `Namespace.IsUser`

[not blocking]

Reply:
Agreed methods seem preferable 👍 

Same [not blocking] comment, I don't think refactoring functions to methods needs to happen in this PR.

Reply:
Yeah I agree with moving to methods